test/views/DeclarationDetailsViewSpec.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import java.time.ZonedDateTime
+
+import scala.collection.JavaConverters.asScalaIteratorConverter
+
+import base.{Injector, OverridableInjector}
+import config.featureFlags._
+import controllers.routes
+import models.declaration.notifications.Notification
+import models.declaration.submissions.Submission
+import models.declaration.submissions.SubmissionStatus.{ACCEPTED, RECEIVED, REJECTED, UNKNOWN}
+import org.mockito.Mockito.when
+import play.api.inject.bind
+import views.declaration.spec.UnitViewSpec
+import views.helpers.StatusOfSubmission
+import views.helpers.TimelineEvents.dateTimeAsShown
+import views.html.declaration_details
+
+class DeclarationDetailsViewSpec extends UnitViewSpec with Injector {
+
+  private val submission = Submission("id", "eori", "lrn", Some("mrn"), Some("ducr"), Seq.empty)
+
+  private val now = ZonedDateTime.now
+
+  private val notifications = List(
```

Comment:
Would be good to test for the scenario where there are no Notifications for a given submission (how would the page handle that?)

Reply:
I thought about that but it seemed to me that we always have at least one notification if we are in that page... but I might be wrong :upside_down_face:

Reply:
According to what business wants, you're right @udinobi . But IMHO we should be covering for unhappy-path scenarios.